lib/pure/os.nim

```diff
@@ -1716,15 +1716,54 @@ proc createSymlink*(src, dest: string) {.noWeirdTarget.} =
     if symlink(src, dest) != 0:
       raiseOSError(osLastError(), $(src, dest))
 
+when defined(windows):
+  proc getFinalPathNameByHandleW(
+    hFile: Handle,
+    lpszFilePath: WideCStringObj,
+    cchFilePath: DWORD,
+    dwFlags: DWORD
+  ): DWORD {.importc: "GetFinalPathNameByHandleW",
+    stdcall, dynlib: "Kernel32.dll".}
+
 proc expandSymlink*(symlinkPath: string): string {.noWeirdTarget.} =
   ## Returns a string representing the path to which the symbolic link points.
-  ##
-  ## On Windows this is a noop, `symlinkPath` is simply returned.
+  ## 
+  ## .. Note:: The function fails if `symlinkPath` is not a symlink, except on Windows, 
+  ##    where the full path will be returned.
```

Comment:
* is that what your implementation does though?
* IMO we should make windows behavior consistent with posix, but can be done in future work (possibly adding a param `checkSymlink = false` or maybe `true`)

Reply:
fixed